File: app/src/main/java/com/alexanderageychenko/ecometer/Tools/dagger2/Module/AppModule.java
package com.alexanderageychenko.ecometer.Tools.dagger2.Module;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Alexander on 15.04.2017.
 */

@Module
public class AppModule {

    Application mApplication;

    public AppModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    Application providesApplication() {
        return mApplication;
    }
}

File: app/src/main/java/com/wackalooon/ecometer/di/AppModule.kt
package com.wackalooon.ecometer.di

import android.content.Context
import com.wackalooon.ecometer.EcoMeterApp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {

    @Provides
    @Singleton
    fun provideContext(application: EcoMeterApp):Context = application
}
